gpt_engineer/ai.py

```diff
@@ -39,7 +39,7 @@ def fassistant(self, msg):
 
     def next(self, messages: list[dict[str, str]], prompt=None):
         if prompt:
-            messages = messages + [{"role": "user", "content": prompt}]
+            messages += [{"role": "user", "content": prompt}]
```

Comment:
Extending a list modifies the argument in python no?

Reply:
I believe both lines do exactly the same.

Reply:
yes they do.

Reply:
They don't, `+=` modifies the variable for the caller, the old code creates a new variable. I was testing this out while you merged lol:

```py
def foo1(messages):
    messages += ["role"]

def foo2(messages):
    messages = messages + ["role"]

m1 = ["m1"]
foo1(m1)
print(m1)
m2 = ["m2"]
foo2(m2)
print(m2)
```

This prints

```
["m1"]
["m2", "role"]
```

Reply:
what python version?

Reply:
https://python-reference.readthedocs.io/en/latest/docs/operators/addition_assignment.html

Reply:
Call them the other way around:

```python
def foo1(messages):
    messages += ["role"]

def foo2(messages):
    messages = messages + ["role"]

m2 = ["m2"]
foo2(m2)
print(m2)

m1 = ["m1"]
foo1(m1)
print(m1)
```

this prints:
```
['m2']
['m1', 'role']
```

Reply:
Oops, copied the wrong output, yeah it was:

```
['m2']
['m1', 'role']
```